celer/tests/test_lasso.py

```diff
@@ -210,7 +210,16 @@ def test_weights_lasso():
 
     assert_allclose(clf1.coef_, clf2.coef_ / weights)
 
+    # support inf weights
```

Comment:
add a separate test, called `test_infinite_weights`. this way when the test fails, you know exactly what went wrong. One big test which tests a lot of different stuff is not easy to debug.